nemo_curator/classifiers/aegis.py

```diff
@@ -0,0 +1,246 @@
+from nemo_curator.utils.aegis_utils import format_aegis
+
+
+@dataclass
+class AegisConfig:
+    peft_model_name_or_path: str
+    token: Optional[Union[str, bool]] = None
+    pretrained_model_name_or_path: str = "meta-llama/LlamaGuard-7b"
+    dtype: torch.dtype = torch.bfloat16
+    max_length: int = 4096
+
+
+AEGIS_LABELS = [
+    "unknown",
+    "safe",
+    "O1",
+    "O2",
+    "O3",
+    "O4",
+    "O5",
+    "O6",
+    "O7",
+    "O8",
+    "O9",
+    "O10",
+    "O11",
+    "O12",
+    "O13",
+]
+
+
+class AegisModel(nn.Module):
+    def __init__(
+        self,
+        pretrained_model_name_or_path: str,
+        peft_model_name_or_path: str,
+        dtype: torch.dtype,
+        token: str,
+    ):
+        super().__init__()
+        base_model = AutoModelForCausalLM.from_pretrained(
+            pretrained_model_name_or_path, torch_dtype=dtype, token=token
+        )
+        self.model = PeftModel.from_pretrained(base_model, peft_model_name_or_path)
+
+    @torch.no_grad()
+    def forward(self, batch):
+        response = self.model.generate(
+            **batch,
+            max_new_tokens=100,
+            pad_token_id=0,
+        )
+        return response
+
+
+class AegisHFModel(HFModel):
+    def __init__(self, config: AegisConfig):
+        self.config = config
+        super().__init__(
+            config.pretrained_model_name_or_path,
+            max_mem_gb=48,
+            start_batch_size=4,
+            end_batch_size=32,
+            batch_size_increment=4,
+            start_seq_len=1024,
+            seq_len_increment=1024,
+        )
+
+    def load_model(self, device="cuda"):
+        model = AegisModel(
+            self.config.pretrained_model_name_or_path,
+            self.config.peft_model_name_or_path,
+            self.config.dtype,
+            self.config.token,
+        )
+        model = model.to(device)
+        model.eval()
+        return model
+
+    def load_config(self):
+        return AutoConfig.from_pretrained(
+            pretrained_model_name_or_path=self.config.pretrained_model_name_or_path,
+            token=self.config.token,
+        )
+
+    @lru_cache(maxsize=1)
+    def load_cfg(self):
+        return AutoConfig.from_pretrained(
+            pretrained_model_name_or_path=self.config.pretrained_model_name_or_path,
+            token=self.config.token,
+        )
+
+    @lru_cache(maxsize=1)
+    def load_tokenizer(self):
+        tokenizer = AutoTokenizer.from_pretrained(
+            pretrained_model_name_or_path=self.config.pretrained_model_name_or_path,
+            token=self.config.token,
+            padding_side="left",
+        )
+        tokenizer.pad_token = tokenizer.unk_token
+
+        return tokenizer
+
+    def max_seq_length(self) -> int:
+        return self.config.max_length
+
+
+class AegisClassifier(DistributedDataClassifier):
+    def __init__(
+        self,
+        aegis_variant="nvidia/Aegis-AI-Content-Safety-LlamaGuard-Defensive-1.0",
```

Comment:
Can you add a docstring and/or documentation somewhere to list the possible variants you can use?